frontend/packages/kubevirt-plugin/src/components/cdi-upload-provider/cdi-upload-provider.tsx

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+import { throttle } from 'lodash';
+import axios, { Canceler } from 'axios';
+import {
+  WatchK8sResource,
+  useK8sWatchResource,
+} from '@console/internal/components/utils/k8s-watch-hook';
+import { RouteModel } from '@console/knative-plugin';
+import { UPLOAD_STATUS } from './consts';
+import { getRouteWebURL } from './helpers';
+
+const routeResource: WatchK8sResource = {
+  kind: RouteModel.kind,
+  namespaced: false,
+  isList: true,
+  optional: true,
+  fieldSelector: `metadata.name=cdi-uploadproxy`,
```

Comment:
you dont you just use the `name` field instead of `fieldSelector` ?
you also dont need `isList`.

`optional` actually does not do anything

Reply:
getting 404 when using `name: "cdi-uploadproxy"`
sometimes, werid?

Reply:
looks like without `isList`, I need to provide a namespace for it to work 100% of the time.
using `name` with `isList` was not reliable aswell, i got wrong results.
After many tries this is the best usage for me so far:
```
const routeResource: WatchK8sResource = {
  kind: RouteModel.kind,
  namespaced: true,
  isList: true,
  fieldSelector: 'metadata.name=cdi-uploadproxy',
};
```